lib/web/websocket/receiver.js

```diff
@@ -314,6 +314,10 @@ class ByteParser extends Writable {
       code = data.readUInt16BE(0)
     }
 
+    if (code !== undefined && !isValidStatusCode(code)) {
+      return { code: 1002, reason: 'Invalid status code', error: true }
+    }
```

Comment:
We could actually also integrate this into the if block above, but probably you consider it to invasive.